hare/eligibility/oracle.go

```diff
@@ -251,56 +240,44 @@ func (o *Oracle) Proof(ctx context.Context, layer types.LayerID, round int32) ([
 
 // Returns a map of all active nodes in the specified layer id
 func (o *Oracle) actives(layer types.LayerID) (map[string]struct{}, error) {
-	sl := roundedSafeLayer(layer, types.LayerID(o.cfg.ConfidenceParam), o.layersPerEpoch, types.LayerID(o.cfg.EpochOffset))
-	safeEp := sl.GetEpoch()
-
-	o.With().Info("safe layer and epoch", sl, safeEp)
-	// check genesis
-	// genesis is for 3 epochs with hare since it can only count active identities found in blocks
-	if safeEp < 3 {
-		return nil, errGenesis
-	}
-
 	// lock until any return
 	// note: no need to lock per safeEp - we do not expect many concurrent requests per safeEp (max two)
 	o.lock.Lock()
 	defer o.lock.Unlock()
 
-	// check cache
-	if val, exist := o.activesCache.Get(safeEp); exist {
-		return val.(map[string]struct{}), nil
+	// we can't read blocks during genesis epochs as there are none
+	if layer.GetEpoch().IsGenesis() {
+		return nil, errGenesis
 	}
```

Comment:
This probably isn't relevant based on what was discussed today, but I'll still point it out.

This PR removes the need to read any blocks, so there's no need to return this error anymore. Everything below should work during genesis.